src/components/Card.js

```diff
@@ -1,41 +1,99 @@
+    //variable to register if current user liked the image or not
+    const currentUserLike = this._likes.find((user) => {
+      return user._id === "f50447686616d1fa985ca0e1";
+    });
+    let didCurrentUserLikeThisCard = currentUserLike ? true : false;
+    if (didCurrentUserLikeThisCard) {
+      // 1. Make a call to network to remove the users like for this card
+      api.unLikeACard(this._cardId).then((data) => {
+        this._likes = data.likes;
+        this._favIconElement.classList.remove("card__fav-icon-selected");
+        this._favCountElement.textContent = this._likes.length;
+      });
+    } else {
+      // 1. Make a call to network to add the users like for this card
+      api.likeACard(this._cardId).then((data) => {
+        this._likes = data.likes;
+        // 2. Update state of the like button and update counter
+        this._favIconElement.classList.toggle("card__fav-icon-selected");
+        this._favCountElement.textContent = this._likes.length;
+      });
+    }
+  };
+
+  _handleDeleteCardFunctionInCardComponent = () => {
+    this.deleteCardPopup = new DeleteCardForm(
+      "#delete-image-confirm-modal",
+      this._handleDeleteCardFormSubmit
+    );
+    this.deleteCardPopup.openModal();
   };
 
-  _handleDelButton = () => {
+  _handleDeleteCardFormSubmit = () => {
     this._cardElement.remove();
-    this._cardElement = null;
+    this._handleCardDeleteFunctionInIndexComponent(this._cardId);
+    if (this.deleteCardPopup) {
+      this.deleteCardPopup.closeModal();
+    }
   };
 
   _onCardClick = (ev) => {
     this._handleCardClick(this._cardData);
   };
 
-  _setEventListeners() {
-    this._favIconElement.addEventListener("click", this._handleFavIcon);
-    this._deleteCardButton.addEventListener("click", this._handleDelButton);
-    this._cardElement
-      .querySelector(".card__image")
-      .addEventListener("click", this._onCardClick);
-  }
-
   getCardElement() {
     this._cardTemplate = document.querySelector(this._cardSelector);
     this._cardElement = this._cardTemplate.content
       .querySelector(".card")
       .cloneNode(true);
-    this._deleteCardButton =
-      this._cardElement.querySelector(".card__del-button");
+    this._deleteCardIcon = this._cardElement.querySelector(".card__del-button");
+    if (this._owner === "f50447686616d1fa985ca0e1") {
+      this._deleteCardIcon.classList.remove("card__del-button-hidden");
+    } else {
+      this._deleteCardIcon.classList.add("card__del-button-hidden");
+    }
     this._favIconElement = this._cardElement.querySelector(".card__fav-icon");
+    const userLikes = this._likes.find((user) => {
+      return user._id === "f50447686616d1fa985ca0e1";
+    });
+    if (userLikes) {
+      this._favIconElement.classList.add("card__fav-icon-selected");
+    } else {
+      this._favIconElement.classList.add("card__fav-icon");
+    }
+    this._favCountElement = this._cardElement.querySelector(".card__fav_count");
```

Comment:
Good job that you've found these variables only once here and defined them as **class fields**